src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts

```diff
@@ -175,27 +181,50 @@ class KubeflowTrainingService implements TrainingService {
             'utf-8'
         );
 
-        // Creat work dir for current trial in NFS directory 
-        await cpp.exec(`mkdir -p ${this.trialLocalNFSTempFolder}/nni/${getExperimentId()}/${trialJobId}`);
-        // Copy code files from local dir to NFS mounted dir
-        await cpp.exec(`cp -r ${trialLocalTempFolder}/* ${this.trialLocalNFSTempFolder}/nni/${getExperimentId()}/${trialJobId}/.`);
-
-        const nfsConfig: NFSConfig = this.kubeflowClusterConfig.nfs;
-        const trialJobDetail: KubeflowTrialJobDetail = new KubeflowTrialJobDetail(
-            trialJobId,
-            'WAITING',
-            Date.now(),
-            trialWorkingFolder,
-            form,
-            kubeflowJobName,
-            curTrialSequenceId,
-            `nfs://${nfsConfig.server}:${path.join(nfsConfig.path, 'nni', getExperimentId(), trialJobId, 'output')}`,
-            this.kubeflowJobPlural
-            );
+        let trialJobDetail: KubeflowTrialJobDetail;
+
+        if(this.kubeflowClusterConfig.nfs){
+            // Creat work dir for current trial in NFS directory 
+            await cpp.exec(`mkdir -p ${this.trialLocalNFSTempFolder}/nni/${getExperimentId()}/${trialJobId}`);
+            // Copy code files from local dir to NFS mounted dir
 
+            
+            await cpp.exec(`cp -r ${trialLocalTempFolder}/* ${this.trialLocalNFSTempFolder}/nni/${getExperimentId()}/${trialJobId}/.`);
+        
+            const nfsConfig: NFSConfig = this.kubeflowClusterConfig.nfs;
+            trialJobDetail = new KubeflowTrialJobDetail(
+                trialJobId,
+                'WAITING',
+                Date.now(),
+                trialWorkingFolder,
+                form,
+                kubeflowJobName,
+                curTrialSequenceId,
+                `nfs://${nfsConfig.server}:${path.join(nfsConfig.path, 'nni', getExperimentId(), trialJobId, 'output')}`,
+                this.kubeflowJobPlural
+            );
+        }else{
+            try{
+                //upload local files to azure storage
+                await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient, `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare, `${trialLocalTempFolder}`);
+                trialJobDetail = new KubeflowTrialJobDetail(
+                    trialJobId,
+                    'WAITING',
+                    Date.now(),
+                    trialWorkingFolder,
+                    form,
+                    kubeflowJobName,
+                    curTrialSequenceId,
+                    `https://${this.azureStorageAccountName}.file.core.windows.net/${this.azureStorageShare}/${path.join('nni', getExperimentId(), trialJobId, 'output')}`,
```

Comment:
add comment to explain what is it?
If possible, can you move the format string into kubeflowData?

Reply:
Most of the code from Line 209 to line 218 is duplicated with line 195 to line 204. suggest to refactor to remove the duplication.

Reply:
Have you judged if this.azureStorageAccountName and this.azureStorageShare is undefined? 

It's dangerous to use 'any' as the type of this.azureStorageAccountName and this.azureStorageShare